src/client/components/TablePaginated/Filters/Country/hooks/useTooltipContent.ts

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { Objects } from 'utils/objects'
+
+import { useTablePaginatedFilterValue } from 'client/store/ui/tablePaginated/hooks'
+
+type Props = {
+  fieldName: string
+  path: string
+}
+
+type Returned = {
+  hideTooltip: () => void
+  showTooltip: () => void
+  tooltipContent: string | null
+}
+
+export const useTooltipContent = (props: Props): Returned => {
+  const { fieldName, path } = props
+  const filterValue = useTablePaginatedFilterValue<Array<string>>(path, fieldName)
+  const [canDisplayTooltip, setCanDisplayTooltip] = useState(true)
+  const { t } = useTranslation()
+
+  const tooltipContent = useMemo<string | null>(() => {
+    if (Objects.isEmpty(filterValue)) return null
+    if (!canDisplayTooltip) return null
+    const MAX_VISIBLE_COUNTRY_LABELS = 50
+
+    const selectedCountryLabels = filterValue
+      .map((countryIso) => {
+        return t(`area.${countryIso}.listName`)
+      })
+      .sort((a, b) => a.localeCompare(b))
+      .slice(0, MAX_VISIBLE_COUNTRY_LABELS)
+
+    if (filterValue.length > MAX_VISIBLE_COUNTRY_LABELS) {
+      selectedCountryLabels.push('...')
+    }
+
+    return selectedCountryLabels.join(', ')
```

Comment:
Possible improvements for the countries tooltip:
1. If all countries in a region are selected, then display the region name instead of its selected countries.
2. Instead of showing "..." at the end when too many are selected, show "+{{remainingCount}} more". For example: "... Somalia, South Africa, +78 more."

Reply:
Yes, I agree a separate PR is better for this 👍